js/damage.js

```diff
@@ -543,6 +543,9 @@ function getDamageResult(attacker, defender, move, field) {
     if (attacker.ability === "Tinted Lens" && typeEffectiveness < 1) {
         finalMods.push(0x2000);
         description.attackerAbility = attacker.ability;
+    } else if (field.isFriendGuard) {
```

Comment:
You have to split Tinted Lens, Friend Guard, and Sniper into three separate `if` statements like most of the others around here; I have no idea why `else if` was used originally. Otherwise Friend Guard won't be calculated with Tinted Lens, and Sniper won't be calculated with Friend Guard.